src/plugins/discover/public/application/apps/main/components/doc_table/components/table_row_details.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiLink, EuiTitle } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+interface TableRowDetailsProps {
+  open: boolean;
+  colLength: number;
+  isTimeBased: boolean;
+  getContextAppHref: () => string;
+  getSingleDocHref: () => string;
+  children: JSX.Element;
+}
+
+export const TableRowDetails = ({
+  open,
+  colLength,
+  isTimeBased,
+  getContextAppHref,
+  getSingleDocHref,
+  children,
+}: TableRowDetailsProps) => {
+  if (open) {
```

Comment:
can we reverse - `if (!open) return null`, so then the majority of the function doesn't have to be indented?

Reply:
Yes, sure.